src/main/java/com/linkedin/kmf/apps/SingleClusterMonitor.java

```diff
@@ -71,11 +73,26 @@ public SingleClusterMonitor(Map<String, Object> props, String name) throws Excep
   @Override
   public void start() {
     _topicManagementService.start();
-    CompletableFuture<Void> completableFuture = _topicManagementService.topicManagementResult();
-    completableFuture.thenRun(() -> {
+    CompletableFuture<Void> topicPartitionResult = _topicManagementService.topicPartitionResult();
+    try {
+      /* Delay 2 second to reduce the chance that produce and consumer thread has race condition
+      with TopicManagementService and MultiClusterTopicManagementService */
+      long threadSleepMs = 1000 * 2;
```

Comment:
Can you use `TimeUnit.SECONDS.toMillis(2);`

Reply:
Done.